src/savegame.cpp

```diff
@@ -635,69 +643,1183 @@ void ingame_savegame::write_game(config_writer &out) {
 	out.close_child("replay");
 }
 
-//changes done during 1.11.0-dev
-static void convert_old_saves_1_11_0(config& cfg)
+std::string convert_version_to_suffix(config& cfg)
 {
-	if(!cfg.has_child("snapshot")){
-		return;
+	std::string suffix = "";
+	version_info loaded_version(cfg["version"]);
+
+	if(loaded_version < version_info("1.5.0") && loaded_version >= version_info("1.3.0"))
+		suffix = "_1_4";
+	if(loaded_version < version_info("1.7.0") && loaded_version >= version_info("1.5.0"))
+		suffix = "_1_6";
+	if(loaded_version < version_info("1.9.0") && loaded_version >= version_info("1.7.0"))
+		suffix = "_1_8";
+	if(loaded_version < version_info("1.11.0") && loaded_version >= version_info("1.9.0"))
+		suffix = "_1_10";
+	if(loaded_version < version_info("1.13.0") && loaded_version >= version_info("1.11.0"))
+		suffix = "_1_12";
+
+	return suffix;
+}
+
+std::string append_suffix_to_unit_type(std::string unit_type, config& cfg)
+{
+	const std::string suffix = convert_version_to_suffix(cfg);
+	return unit_type + suffix;
+}
+
+void update_start_unit_types(config& replay_start, config& cfg)
+{
+	const std::string suffix = convert_version_to_suffix(cfg);
+	if(not suffix.empty())
```

Comment:
We prefer to use symbols rather than names for operators; in this case it should be `!`. Not sure if there are other instances of this; see iso646.h for a list of search terms to replace.